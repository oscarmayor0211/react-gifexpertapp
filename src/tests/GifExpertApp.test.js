import GifExpertApp from '../GifExpertApp';
import { shallow } from 'enzyme'
import React from 'react';

describe('Prueba con el <GifExpertApp />', () => {
    
    test('debe de mostrar  <GifExpertApp /> correctamente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect( wrapper ).toMatchSnapshot();
    }),

    test('debe de mostrar una lista de categorias ', () => {

        const categories = ['One unch', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories = {categories} />);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length);

        
    })
    
    
})
