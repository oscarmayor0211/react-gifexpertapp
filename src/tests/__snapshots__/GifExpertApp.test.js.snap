// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba con el <GifExpertApp /> debe de mostrar  <GifExpertApp /> correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul />
</Fragment>
`;

exports[`Prueba con el <GifExpertApp /> debe de mostrar una lista de categorias  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ul>
    <GifGrid
      category="One unch"
      key="0"
    />
    <GifGrid
      category="Dragon Ball"
      key="1"
    />
  </ul>
</Fragment>
`;
